src/encoded/static/components/award.js

```diff
@@ -1690,97 +1790,60 @@ const ExperimentDate = (props) => {
         );
     }
 
-    function fillDates(sortedArray, deduplicated, awardStartDate) {
-        const fillArray = [];
-
-        // Add an object with the award start date to both arrays
-        sortedArray.unshift({ key: awardStartDate, doc_count: 0 });
-
-        // Add objects to the array with doc_count 0 for the missing months
-        const sortedTermsLength = sortedArray.length;
-        for (let j = 0; j < sortedTermsLength - 1; j += 1) {
-            fillArray.push(sortedArray[j]);
-            const startDate = moment(sortedArray[j].key);
-            const endDate = moment(sortedArray[j + 1].key);
-            const monthdiff = endDate.diff(startDate, 'months', false);
-            if (monthdiff > 1) {
-                for (let i = 0; i < monthdiff; i += 1) {
-                    fillArray.push({ key: startDate.add(1, 'months').format('YYYY-MM'), doc_count: 0 });
-                }
-            }
-        }
-        fillArray.push(sortedArray[sortedArray - 1]);
-
-        // Remove any objects with keys before the start date of the award
-        const shortenedArray = [];
-        for (let j = 0; j < fillArray.length - 2; j += 1) {
-            if (moment(fillArray[j].key).isSameOrAfter(awardStartDate, 'date')) {
-                shortenedArray.push(fillArray[j]);
-            }
-        }
-        const formatTerms = shortenedArray.map((term) => {
-            const formattedDate = moment(term.key, ['YYYY-MM']).format('MMM YY');
-            return { key: formattedDate, doc_count: term.doc_count };
-        });
-        // Deduplicate dates
-        formatTerms.forEach((elem) => {
-            if (deduplicated[elem.key]) {
-                deduplicated[elem.key] += elem.doc_count;
-            } else {
-                deduplicated[elem.key] = elem.doc_count;
-            }
-        });
-        return (deduplicated);
-    }
-
-    function createDataset(deduplicated, accumulatorType, cumulativeData) {
-        let cumulativedataset = cumulativeData;
-        let accumulator = accumulatorType;
-        // Create an array of data from objects' doc_counts
-        const dataset = Object.keys(deduplicated).map(item => deduplicated[item]);
+    function createDataset(deduplicated) {
+        let accumulator = 0;
         // Make the data cumulative
-        const accumulatedData = dataset.map((term) => {
-            accumulator += term;
-            cumulativedataset = accumulator;
-            return cumulativedataset;
+        const accumulatedData = deduplicated.map((term) => {
+            accumulator += term.doc_count;
+            return accumulator;
         });
```

Comment:
You could do `accumulatedData = deduplicated.reduce((accumulator, term) => accumulator + term.doc_count)`, but the current way is probably more readable.

Reply:
I _love_ reduce(), but accumulatedData is an array with the same number of elements as deduplicated, and completely based on deduplicated’s contents. It’s used to display the y-axis values over time in the cumulative chart.

Reply:
Ah, righto. I was reading too fast.